plugins/serializers/prometheus_remote_write/prometheus_remote_write.go

```diff
@@ -0,0 +1,341 @@
+package prometheus_remote_write
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/influxdata/telegraf/plugins/serializers/prometheus"
+	"hash/fnv"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+type MetricKey uint64
+
+// MetricSortOrder controls if the output is sorted.
+type MetricSortOrder int
+
+const (
+	NoSortMetrics MetricSortOrder = iota
+	SortMetrics
+)
+
+// StringHandling defines how to process string fields.
+type StringHandling int
+
+const (
+	DiscardStrings StringHandling = iota
+	StringAsLabel
+)
+
+type FormatConfig struct {
+	MetricSortOrder MetricSortOrder
+	StringHandling  StringHandling
+}
```

Comment:
should these be configurable by the user?

Reply:
`StringHandling` is already configurable with `PrometheusStringAsLabel` 
`MetricSortOrder` is present only for test to avoid diff after deserialize.  Prometheus remote write format isn't human readable, so no need to sort for debug